run-ib-pr-matrix.sh

```diff
@@ -26,7 +26,7 @@ if [ "${CHECK_WORKFLOWS}" = "true" ] ; then
     high_stats ) ;;
     nano ) OPTS="-w nano" ;;
     * ) if is_in_array "${TEST_FLAVOR}" "${ALL_GPU_TYPES[@]}" ; then
-          OPTS="-w gpu"
+          OPTS="-w gpu --gpu required"
```

Comment:
also `function is_in_array` is duplicate of the one in `_helper_functions.sh` . so once you source `_helper_functions.sh` here then remove the  `function is_in_array`  from this file too